### PR TITLE
Docs for Decoupled TaintManager from NodeLifeCycleController (KEP-3902)

### PR DESCRIPTION
**Pull request by atosatto**
Placeholder docs for [KEP-3902](https://github.com/kubernetes/enhancements/pull/3901)
Implementation on-going in https://github.com/kubernetes/kubernetes/pull/119208

### CODE REVIEW

**Comment by drewhagen** [comment]
Hello @atosatto @yuanchen8911 :wave:, I see there are some changes to the docs in progress. 👍  Please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday 14th November 2023. Thank you!

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -189,6 +189,7 @@ For a reference to old feature gates that are removed, please refer to
 | `SchedulerQueueingHints` | `false` | Alpha | 1.28 | |
 | `SecurityContextDeny` | `false` | Alpha | 1.27 | |
 | `SidecarContainers` | `false` | Alpha | 1.28 | |
+| `SeparateTaintEvictionController` | `true` | Beta | 1.29 | |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:192]: Move this entry above the previous one for ordering's sake?

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -702,6 +703,7 @@ Each feature gate is designed for enabling/disabling a specific feature:
   for all workloads.
   The seccomp profile is specified in the `securityContext` of a Pod and/or a Container.
 - `SecurityContextDeny`: This gate signals that the `SecurityContextDeny` admission controller is deprecated.
+- `SeparateTaintEvictionController`: Enables running `TaintEvictionController` - that performs [Taint Based Evictions](/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions) - in a separated controller from `NodeLifecycleController`.
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:706]: Please wrap the long lines manually.
Maybe we can add a sentence or two articulating the user perceivable impacts from this.
As a user I need to know whether/when to turn this gate off.
```suggestion
- `SeparateTaintEvictionController`: Enables running `TaintEvictionController`
  that performs [Taint Based Evictions](/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions)
  in a separated controller from `NodeLifecycleController`.
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -726,6 +727,11 @@ Each feature gate is designed for enabling/disabling a specific feature:
   for all workloads.
   The seccomp profile is specified in the `securityContext` of a Pod and/or a Container.
 - `SecurityContextDeny`: This gate signals that the `SecurityContextDeny` admission controller is deprecated.
+- `SeparateTaintEvictionController`: Enables running `TaintEvictionController`,
+  that performs [Taint Based Evictions](/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions),
```
Comment (yuanchen8911) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:731]:
```suggestion
  that performs [Taint-based Evictions](/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions),
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 4]
```diff
@@ -726,6 +727,11 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
+  in a separated controller from `NodeLifecycleController`. When this feature is
+  enabled, users can optionally disable Taint Based Eviction setting the
```
Comment (yuanchen8911) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:733]:
```suggestion
  enabled, users can optionally disable Taint-based Eviction by setting the
```

**Review by yuanchen8911 — APPROVED** [review]
/lgtm

**Review by yuanchen8911 — APPROVED** [review]
/lgtm

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 5]
```diff
@@ -190,6 +190,7 @@ For a reference to old feature gates that are removed, please refer to
 | `ServiceAccountTokenPodNodeInfo` | `false` | Alpha | 1.29 | |
 | `SidecarContainers` | `false` | Alpha | 1.28 | 1.28 |
 | `SidecarContainers` | `true` | Beta | 1.29 | |
+| `SeparateTaintEvictionController` | `true` | Beta | 1.29 | |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:193]: This entry should not appear here. Please make sure the rows are still in alphabetic order.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 6]
```diff
@@ -726,6 +727,11 @@ Each feature gate is designed for enabling/disabling a specific feature:
   for all workloads.
   The seccomp profile is specified in the `securityContext` of a Pod and/or a Container.
 - `SecurityContextDeny`: This gate signals that the `SecurityContextDeny` admission controller is deprecated.
+- `SeparateTaintEvictionController`: Enables running `TaintEvictionController`,
+  that performs [Taint-based Evictions](/docs/concepts/scheduling-eviction/taint-and-toleration/#taint-based-evictions),
+  in a separated controller from `NodeLifecycleController`. When this feature is
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:732]:
```suggestion
  in a controller separated from `NodeLifecycleController`. When this feature is
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 7]
```diff
@@ -726,6 +727,11 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
+  enabled, users can optionally disable Taint-based Eviction setting the
+  `--controllers=-taint-eviction-controller` flag in Kube Controller Manager.
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:734]:
```suggestion
  `--controllers=-taint-eviction-controller` flag on the `kube-controller-manager`.
```

**Comment by drewhagen** [comment]
Hello @atosatto @bradtopol @reylejano :wave:, 

Friendly reminder that the deadline to have docs merged is next Tuesday 28th November 2023.  If this feature needs documentation and the docs are not ready, the feature may be removed from the milestone.

Let me know if you have any questions. Thank you!

**Pull request merged**